Kernel/TermTransformer.hpp

```diff
@@ -38,14 +38,25 @@ namespace Kernel {
  */
 class TermTransformer {
 public:
+  TermTransformer(bool shared = true, bool recurseIntoReplaced = false) : 
+    _sharedResult(shared), _recurseIntoReplaced(recurseIntoReplaced) {}
   virtual ~TermTransformer() {}
   Term* transform(Term* term);
   Literal* transform(Literal* lit);
+  TermList transform(TermList ts);
 protected:
   virtual TermList transformSubterm(TermList trm) = 0;
   Term* transformSpecial(Term* specialTerm);
-  TermList transform(TermList ts);
+  // currently this can be used to transform the
+  // first-order / "green" subterms of an applicative term.
+  // TODO update to allow transformation of prefix subterms
+  // TODO try and code share with transform(Term*)
```

Comment:
I think this should definitely be done before merging. It looks like a big bunch of complicatied code that needs maintaining and can diverge otherwise.

Reply:
It is actually a big simplification in that it allowed me to get rid of three other big blobs of code.

Any suggestions on how to code share with `transform(Term*)`. The two are almost identical, but I don't want to merge them into one, in order to avoid slowing down `transform(Term*)`